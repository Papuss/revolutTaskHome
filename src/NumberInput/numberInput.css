.number-component .number-input {
    display: flex;
}

.number-component .number-input .input {
    border: none;
    outline: none;
    background-color: lightgray;
}
