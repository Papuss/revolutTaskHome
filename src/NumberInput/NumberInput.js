import "./numberInput.css";
import MaskedInput from "react-text-mask";
import createNumberMask from "text-mask-addons/dist/createNumberMask";
import React from 'react';

function NumberInput(props) {
    return (
        <div className="number-component" id={props.id}>
            <div className="number-input">
                <MaskedInput
                    mask={createNumberMask({
                        prefix: "",
                        allowDecimal: true,
                        allowNegative: false,
                        thousandsSeparatorSymbol: "",
                        decimalSymbol: "."
                    })}
                    value={props.value}
                    onChange={event => {
                        props.onChange(event.target.value);
                    }}
                    type="text"
                    className="input"
                    placeholder="0.00"
                />

            </div>
        </div>
    )
}

export default NumberInput;