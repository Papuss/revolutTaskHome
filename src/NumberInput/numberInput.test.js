import NumberInput from './NumberInput';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow,mount,render } from 'enzyme';
import React from 'react';

Enzyme.configure({ adapter: new Adapter() });

describe('Renders NumberInput component', () => {
it('renders as expected', () => {    
    const onChange = jest.fn();
    const wrapper = shallow(
        <NumberInput 
            id="to"
            value=""
            onChange={() => {onChange()}}
        />
    );

    expect(wrapper).toMatchSnapshot();
});
});
