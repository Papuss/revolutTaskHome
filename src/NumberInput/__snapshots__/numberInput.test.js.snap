// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NumberInput component renders as expected 1`] = `ShallowWrapper {}`;
